experiments/modern/src/App.js

```diff
@@ -213,23 +219,38 @@ function XmlNode({ node }) {
     }
     return null;
   }
-  return <Component {...attributes}>{children}</Component>;
+  return <Component node={node} {...attributes}>{children}</Component>;
 }
 
 function App() {
   const [data, setData] = React.useState(null);
   React.useEffect(() => {
-    getSkin().then(setData);
+    getSkin().then(async ({ root, registry }) => {
+      // Execute scripts
+      await Utils.asyncTreeFlatMap(root, async node => {
+        switch (node.xmlNode.name) {
+          case "script": {
+            const scriptGroup = Utils.findParentNodeOfType(node, ["Group", "WinampAbstractionLayer"]);
+            const system = new System(scriptGroup);
```

Comment:
Interesting! I was imagining that there was a single `System` instance shared between all scripts. Is this just part of the hack, or is there some reason to believe we should really have `n` System objects?

Reply:
Yea, at first I was tempted to make the `System` object just the root `MakiObject for the `WinampAbstractionLayer` node. As far as I can tell, all the `System` objects are the same minus one key difference, they each have their own `getScriptGroup()` which will return a different `MakiObject` based on where the script is included.

Maybe there is a way to have them all be the same object, but just return different values for that, but seemed simpler to just try hacking it with different ones to start hah